### DISCUSSION TITLE
Writing to pyproject.toml file so that it processes correctly on pypi

### DISCUSSION THREAD

**Discussion by pkopparla** [PyO3/maturin#737] [Q&A] [answered]
Hello,

I was having a wonderful time with maturin writing python bindings for a rust library, till I hit the publication part.
I'm trying to figure out how to write my `pyproject.toml` file so that pypi shows the correct homepage url, license and other things. It has been incredibly frustrating since there's no examples and the error messages are somewhat vague. For instance, I want my github homepage url to show up on the pypi package page, I've tried both `url = "https://github.com/zonebuilders/zonebuilder-py"` and `homepage="https://github.com/zonebuilders/zonebuilder-py"`, the distribution build works with either but the homepage link doesn't show up in pypi. I think the key needs to be something like `project-url` with some two value array going by the PEP 621 page here: https://www.python.org/dev/peps/pep-0621/#urls

Similar for the license. I can get it to build with `license_file=LICENCSE`, but it doesn't show up on pypi. I've tried many variations with the `license` keyword, eg. `license = ["file", "LICENCE"]`, `license = ["file = LICENCE"]`, `license = ["file" : "LICENCE"]` etc etc and I'm about to lose my mind. How does one figure out the right way to do this? Why are the references are so complicated?

**Comment by messense** [comment, marked as answer]
There is an example in PEP 621: https://www.python.org/dev/peps/pep-0621/#example

**Reply by messense** [reply]
You can also just specify `homepage` and `license` in Cargo.toml, maturin will use it automatically.

**Reply by messense** [reply]
Copied the PEP 621 example to user guide so it's easier to find: https://maturin.rs/metadata.html

**Reply by pkopparla** [reply]
That's great!

**Comment by pkopparla** [comment]
Wonderful, that solves everything. Thank you so much for your help!
